### ISSUE TITLE
feat(embeddings): ColIdefics3 and ColQwen2.5 late-interaction embedders (one 128-dim vector per token) with MaxSim

### ISSUE THREAD

**Issue by inureyes** [lablup/mlxcel#1337]
## Summary

ColIdefics3 (`model_type: idefics3`, `architectures: ["ColIdefics3"]`) and ColQwen2.5 (`model_type: qwen2_5_vl`, `architectures: ["ColQwen2_5"]`, or the native `model_type: colqwen2`, `architectures: ["ColQwen2ForRetrieval"]`) are visual-document retrievers: the VLM runs causally as usual, a linear projection maps every token's final hidden state to 128 dimensions, each token vector is L2-normalized, and similarity is MaxSim over token sets. mlxcel runs both backbones (SmolVLM, Qwen2.5-VL) for generation only. This issue adds both as `EmbeddingModel` implementations with `multi_vector() == true`, the `[num_real_tokens, 128]` response shape defined in #1353, and a MaxSim helper used by `mlxcel embed`.

## Current behavior

- `src/models/detection.rs:335` maps `idefics3` to `ModelType::SmolVLM` and line 319 maps `qwen2_5_vl` to `ModelType::Qwen25VL`; `colqwen2` is unsupported. Both loaders require an `lm_head` (`SmolVLMModel.text_model: Llama3Model`, `src/vision/smolvlm.rs:158-160`, and `Qwen2VLModel.lm_head: UnifiedLinear`, `src/models/qwen2_vl.rs:513`) and return logits.
- `Qwen25VLModel::get_input_embeddings` (`src/vision/qwen2_5_vl.rs:46-120`) already merges image features and computes M-RoPE position ids (`compute_rope_index`), and `Qwen2VLModel::forward_for_sequence` (`src/models/qwen2_vl.rs:627-690`) already broadcasts text-only positions to `[3, B, L]`, so batched multi-row prefill is supported; only a hidden-state exit is missing.
- `SmolVLMModel::get_input_embeddings` (`src/vision/smolvlm.rs:187`) merges SigLIP features through pixel-shuffle and the connector.

### Output contract

`embed` returns `[B, L, 128]` with rows of padding tokens zeroed. The HTTP layer (#1353) trims each item to its real length and returns `embedding: [[...128 floats...], ...]` (`shape: [n, 128]` in base64 mode). `maxsim(q: [Lq, D], d: [Ld, D]) = sum_i max_j dot(q_i, d_j)`, added as `crate::embeddings::maxsim` and printed by `mlxcel embed` for multi-vector models in place of the cosine matrix.

### ColIdefics3

Config (verified on `vidore/ColSmolVLM-Instruct-256M-base`): `text_config` Llama (`hidden_size 576`, 30 layers, `tie_word_embeddings false`), `vision_config` Idefics3 (`hidden_size 768`, 12 layers, `image_size 512`, `patch_size 16`), `scale_factor 4`, `image_token_id 49190`, `embedding_dim 128` (default when absent), `mask_non_image_embeddings false` (true is rejected at load). No `lm_head` in the checkpoint.

Forward:
```
embeds = SmolVLMModel::get_input_embeddings(input_ids, pixel_values)   # or embed_tokens for text
mask = create_causal_padding_mask(attention_mask, 0)                     # right padding
h = llama.forward_hidden(input_ids, Some(embeds), fresh caches, Some(mask))   # norm applied, no head
p = linear(h)                                                             # [B, L, 128], weight [128, 576] + bias
p = p / max(||p||_2 per token, 1e-9); p = p * attention_mask[..., None]
```
Prompt formats (the processor convention for this family): an image document is `<|im_start|>User:<image>Describe the image.<end_of_utterance>\nAssistant:` with `<image>` expanded by the SmolVLM processor into the tile/global image token blocks; a query is `Query: {text}` followed by ten `<end_of_utterance>` augmentation tokens. Both are produced by `format_text` (image vs text item).

Weight keys (verified): `linear.{weight,bias}`, `model.text_model.embed_tokens.weight`, `model.text_model.layers.{i}.*`, `model.text_model.norm.weight`, `model.vision_model.*`, `model.connector.modality_projection.proj.weight`. Sanitize: the SmolVLM mapping already used by `src/loading/vlm_smolvlm.rs`, plus: when `1_Dense/model.safetensors` exists its `linear.{weight,bias}` (exposed as `1_Dense.linear.*`) replace the root `linear.*`; drop `lm_head.*`. LoRA-only repositories (`adapter_model.safetensors` without `model.safetensors`) are rejected with "merge the adapter into the base checkpoint first".

### ColQwen2.5

Config (verified on `vidore/colqwen2.5-base`): the Qwen2.5-VL 3B layout (`hidden_size 2048`, `tie_word_embeddings true`, `sliding_window 32768` unused), `embedding_dim 128` (default). The native layout (`model_type: colqwen2`) carries `embedding_dim: 128` explicitly.

Forward:
```
embeds, position_ids = Qwen25VLModel::get_input_embeddings(input_ids, pixel_values, grid_thw)   # M-RoPE from compute_rope_index
mask = create_causal_padding_mask(attention_mask, 0)
h = qwen2vl.forward_hidden(input_ids, Some(embeds), fresh caches, Some(mask))   # norm applied, no head
p = custom_text_proj(h)                                                          # [B, L, 128], weight [128, 2048] + bias
p = p / max(||p||_2 per token, 1e-9); p = p * attention_mask[..., None]
```
Text-only rows use the sequential `[3, B, L]` positions the backbone already builds. Image inputs keep the real M-RoPE positions (do not substitute sequential positions for image rows).

Prompt formats: image document `<|im_start|>user\n<|vision_start|><|image_pad|><|vision_end|>Describe the image.<|im_end|><|im_start|>assistant\n` (the `<|image_pad|>` run expanded by the Qwen2.5-VL processor, image resized to at most `768 * 28 * 28` pixels); query `Query: {text}` plus ten `<|endoftext|>` augmentation tokens.

Weight keys: `custom_text_proj.{weight,bias}`, `model.embed_tokens.weight`, `model.layers.{i}.*`, `model.norm.weight`, `visual.*` (verified on `vidore/colqwen2.5-base`). Sanitize for the native layout: `embedding_proj_layer.` -> `custom_text_proj.`; `vlm.model.language_model.` -> `model.`; `vlm.model.visual.` -> `visual.`; `vlm.model.` -> `model.`; strip a remaining `vlm.`; then the existing Qwen2.5-VL sanitize; drop `lm_head.*` when tied.

## Implementation plan

1. `src/models/qwen2_vl.rs`: `pub(crate) fn forward_hidden(&self, input_ids, input_embeddings, caches, mask, seq_id) -> [B, L, D]` split out of `forward_for_sequence` (everything through `norm`), `forward_for_sequence` calls it then `lm_head`; byte-identical. `// Used by: Qwen2-VL / Qwen2.5-VL generation, ColQwen25`.
2. `src/models/llama3.rs`: `forward_hidden` and `from_weights_without_head` as specified in #1325 (land whichever merges first; the other reuses it).
3. `src/embeddings/maxsim.rs`: `maxsim(q, d) -> f32` over `Vec<Vec<f32>>` plus an MLX version over `[Lq, D] x [Ld, D]` (`matmul`, `max_axis(-1)`, `sum_all`); tests.
4. `src/models/colidefics3.rs` (+ `_tests.rs`): `ColIdefics3Model { vlm: SmolVLMModel-without-head (vision_model, connector, processor, text backbone), linear: UnifiedLinear, image_token_id, embedding_dim }`; `load(dir)` through the SmolVLM loader pieces with the `1_Dense` override; `impl EmbeddingModel` (`multi_vector = true`, `supports_images = true`, `embedding_dim = 128`, `normalize = true` per token, `format_text` as above).
5. `src/models/colqwen2_5.rs` (+ `_tests.rs`): `ColQwen25Model { vlm: Qwen25VLModel (lm_head tied, unused), custom_text_proj: UnifiedLinear, embedding_dim }`; `load(dir)` through `src/loading/vlm_qwen.rs` with the sanitize above; `impl EmbeddingModel` as for ColIdefics3.
6. `src/models/detection.rs`: in `is_embedding_checkpoint`, `idefics3` with `ColIdefics3` -> `ModelType::ColIdefics3`; `qwen2_5_vl` with `ColQwen2_5`, or `colqwen2` -> `ModelType::ColQwen25`. `src/model_metadata.rs` entries; `src/embeddings/loader.rs` arms.
7. `src/commands/embed.rs`: print the MaxSim matrix for multi-vector models; `docs/embeddings.md` section on multi-vector responses and scoring; `docs/supported-models.md` rows.

## Validation

(a) Unit tests:
- `src/embeddings/maxsim_tests.rs`: `maxsim_identity_equals_query_length` (unit vectors against themselves), `maxsim_is_asymmetric`, `mlx_maxsim_matches_cpu`.
- `src/models/colidefics3_tests.rs`: `sanitize_prefers_1_dense_projection`, `rejects_mask_non_image_embeddings_true`, `query_format_appends_ten_augmentation_tokens`, `padding_rows_are_zero_and_real_rows_unit_norm` (random 2-layer text config, batch with one padded row), `causal_prefill_is_causal` (96 tokens).
- `src/models/colqwen2_5_tests.rs`: `native_layout_keys_remap_to_base_layout` (all five rules), `forward_hidden_then_head_matches_forward` (random 2-layer config, token-exact), `batched_text_rows_match_single_rows` (the `[3, B, L]` positions).

(b) Real checkpoints:
```bash
# ColSmol: merge the LoRA adapter of vidore/colSmol-256M into vidore/ColSmolVLM-Instruct-256M-base with the
# checkpoint's own tooling (or use a merged export), keeping 1_Dense/model.safetensors next to config.json.
./target/release/mlxcel embed -m models/colSmol-256M-merged \
  -p "Query: What was the total revenue in 2023?" \
  --image /path/to/revenue-table.png --image /path/to/unrelated-photo.png
# ColQwen2.5: vidore/colqwen2.5-v0.2 is a LoRA adapter on vidore/colqwen2.5-base; merge it the same way.
./target/release/mlxcel embed -m models/colqwen2.5-v0.2-merged \
  -p "Query: What was the total revenue in 2023?" \
  --image /path/to/revenue-table.png --image /path/to/unrelated-photo.png
./target/release/mlxcel-server -m models/colqwen2.5-v0.2-merged --port 8080
curl -s localhost:8080/v1/embeddings -H 'Content-Type: application/json' -d '{
  "input": [{"type": "text", "text": "Query: What was the total revenue in 2023?"},
            {"type": "image_url", "image_url": {"url": "data:image/png;base64,<revenue table>"}}]}'
```
Acceptance per model: each item returns `[n, 128]` with every row unit-norm within 1e-5 and `n` equal to its real token count (`usage.prompt_tokens` sums them); MaxSim(query, revenue table) exceeds MaxSim(query, unrelated photo) by at least 10% of the larger value; the query embedded alone equals its rows inside a padded batch within 1e-3; `vidore/colqwen2.5-base` (untrained projection) loads and produces finite output, which validates the weight layout independently of the merge step.

## Acceptance criteria

- [x] `ColIdefics3Model` and `ColQwen25Model` load from the base and native layouts (with the `1_Dense` override and the `vlm.`/`embedding_proj_layer` remaps), run the causal stack without `lm_head`, project to 128 dims and normalize per token
- [x] `/v1/embeddings` returns `[n, 128]` per item for these models and `mlxcel embed` prints MaxSim
- [x] the two merged checkpoints pass the gates above
- [x] docs/supported-models.md updated
- [x] detection table in src/models/detection.rs updated with a test
- [x] the equivalent CUDA suites pass on the validation host: `cargo test --profile test-fast --features cuda --lib models::col` and `--lib embeddings::`, both with `--test-threads=1` (metal,accelerate is a macOS gate and is left to CI)
- [x] `cargo clippy --profile test-fast --features cuda --lib --bins --tests -- -D warnings` and `cargo fmt --all -- --check` pass

## Out of scope

- Merging LoRA adapters inside mlxcel.
- ColPali (PaliGemma backbone) and ColQwen2 (Qwen2-VL backbone) variants.
- A retrieval or scoring HTTP endpoint over stored multi-vector documents.

## Dependencies

Blocked by #1353

- Part of epic #1348
